src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs

```diff
@@ -63,10 +64,18 @@ public sealed class BooleanType : AtomicType
     }
 
     /// <summary>
-    /// Represents a date type.
+    /// Represents a date type. It represents a valid date in the proleptic Gregorian
+    /// calendar. Valid range is [0001-01-01, 9999-12-31].
     /// </summary>
     public sealed class DateType : AtomicType
     {
+        internal override bool NeedConversion() => true;
+
+        internal override object FromInternal(object obj)
+        {
+            var unixTimeEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```

Comment:
do we need to worry about timezone here?

Reply:
Nope, I have removed timezone.